src/update_eh.cpp

```diff
@@ -164,15 +165,19 @@ bool fields_chunk::update_eh(field_type ft, bool skip_w_components) {
                sizeof(realnum) * gv.ntot());
       }
 
-      if (f[ec][cmp] != f[dc][cmp])
-        STEP_UPDATE_EDHB(f[ec][cmp], ec, gv, gv.little_owned_corner(ec), gv.big_corner(),
-                         dmp[dc][cmp], dmp[dc_1][cmp], dmp[dc_2][cmp],
-                         s->chi1inv[ec][d_ec], dmp[dc_1][cmp] ? s->chi1inv[ec][d_1] : NULL,
-                         dmp[dc_2][cmp] ? s->chi1inv[ec][d_2] : NULL, s_ec, s_1, s_2, s->chi2[ec],
-                         s->chi3[ec], f_w[ec][cmp], dsigw, s->sig[dsigw], s->kap[dsigw]);
+      if (f[ec][cmp] != f[dc][cmp]) {
+        for (const auto& sub_gv : gvs) {
```

Comment:
1. This loop should be outside the `FOR_FT_COMPONENTS` loop (just make sure to do the backup copies only once).
2. If the material has a diagonal `s->chi1inv` (i.e. the off-diagonal pointers are `NULL` for this `ft`), then you *don't* want to tile (because there is no locality as explained below) — in that case you want to just set `gvs` to a single tile with the whole `gv`.